ibmcloud/image/build.sh

```diff
@@ -53,7 +65,15 @@ fi
 
 function cleanup () {
     msg=$1
-
+    if [ "${SE_BOOT}" = "1" ]; then
+        for mnt in "$dst_mnt/boot-se" "$dst_mnt/etc/keys" "$dst_mnt/sys"; do
+            mountpoint -q "$mnt" && umount "$mnt" || true
+            [[ -d "$mnt" ]] && rmdir "$mnt" 2> /dev/null || true
+        done
+        [ -n "${LUKS_NAME:-}" ] && cryptsetup close $LUKS_NAME 2> /dev/null || true
```

Comment:
After doing some failing testcases and running the script with the hostkey not in the correct directory it fails on the `genprotimg` step at line 347 (to be expected).

However it seems that the closing of the partition for `tmp_nbd` is not correct. When it finishes currently that partition stays open, and means the next build will fail until either it is:
1. Manually closed after the script fully finishes by doing: 
```
qemu-nbd --disconnect "/dev/nbd1/"
export LUKS_NAME="LUKS-$(blkid -s UUID -o value nbd1p2)"
cryptsetup close ${LUKS_NAME}
```
2. Or manually changing the `tmp_nbd` to point at a different instance. e.g. `/dev/nbd2` inside the `build.sh` file. 

I did find a **potential solution** of changing the line to be:
```
[ -n "${LUKS_NAME:-}" ] && (cryptsetup close $LUKS_NAME 2> /dev/null || true)
```
And moving the line to Line: [build.sh#L83](https://github.com/liudalibj/cloud-api-adaptor/blob/c3c684c9b9b54caf389ebcf78de58291d51c35df/ibmcloud/image/build.sh#L83)
I think this happens because the `||`  is seen as the other response of the inline statement. And the partition is still seen as being in use. After moving the line too after the other unmounts it seems to successfully remove them. 
I have a feeling that this is something to do with `${dst_mnt}` not properly unmounted until after that loop. So it cant successfully close those partitions. 


It might also be worth creating some form of troubleshoot area at the end of the file to provide context of the error and a potential solution if this happens again.
The error we see with this when attempting to run the `make build` again is: 
```
Formatting './tmp.qcow2', fmt=qcow2 size=107374182400 cluster_size=65536 lazy_refcounts=off refcount_bits=16
Failed to set NBD socket
Disconnect client, due to: Unexpected end-of-file before all bytes were read
```
And the checks to see if it related to the partition issue is to run `ls /dev/nbd1p*` which should result in:
```
/dev/nbd1p1
/dev/nbd1p2
```
If the error has occurred.

Reply:
Refine the build.sh to handle the bad path and add one Troubleshooting section to the document.